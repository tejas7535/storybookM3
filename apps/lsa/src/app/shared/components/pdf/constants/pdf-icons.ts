export const starsIconPrimary =
  'iVBORw0KGgoAAAANSUhEUgAAAGAAAABgCAYAAADimHc4AAAAAXNSR0IArs4c6QAABqpJREFUeF7tXVty3DYQJJVrxbF9qbjyFevL5VzK8uNa1qaAXaxAimT3vACoivxxWcJiBt0zPQOSws7TeXVFYO5q/TQ+nQR0DoKTgJOAzgh0Nn9mwElAZwQ6mz8z4CQAIPDlzw95xMPD9d/58v7+ict0+9n0lH92mb8vZnt+fpr++XX93aDXeBlQAP9j/ncqAFvBm6enTM6nH5+tU3l/fgwCEugJ8BzFt6j2XmmZr5AxSHb0JaAAHw36IZnzY8/M6EPAEMBXrHSUqLYEjAb8q8yYH6fG0tSGgOGBXzPRTpbiCUjgP8zfompq7LzxRMQS8PWvz9N0uXY32ivpc+nxkzzUV93jr/cL1w/ZbOc5YkmII8ACfgL99+Uxr9+6kcoZ+PAhb+DU3VYcCTEE/Pfum3ixBXQr4CjTtIGR/Pv750c0vfT3vgSoi21chO0CoiXi+fLRnJWLDlhK2dH4r+8usukcgE+kW7JGSoRzJvhlgER2vOQmgefVt0v9d5IjHwJEUeQQ9SXNEmjpcgJj6rAOOwEdnL7LXJG8Tz/t6yiTNl6PzXGJs87FaxGtkXOjoma0bSSALLpGJzcxqMl3LozZnmQHb1ifngA6+h01v2Zi3XF5ylCxw5JgCAAdAaxjUdv4LfINUXioMmygKe3rCKCcCor8hNaWfUMUIpmnuiOlfTkBDPhKZyAQ905lp/ZEyFDd8qJ7SYoskBPAbFgigTiSPwUANOmM7CoCT0YAE/1Rus/06QoAaAL2pG89gTAIhAQQbWdk9GcQDnyIJgDZT78X+sATwKRgdPQzAAgjUJQBAVnAE8DIT3j0E0/YoglIJKA6KMgCngBktEX0Qx/kEiDOAGaXHEIAutcfHf2M/BQ0R/CFzEQuA5D8CBhXRRyrvWVycvFqXxgZIn3gCICpH7jrZdrPNZItAgI1JaQPHAFIfki2TRGHfFhPHi1DiIDkD+EDJkBrqLynY0L99uH8twHSd3xurxl62E9zbD13RspABCYmAOn/XvejfkPCCzGneY6eXw9NgEa7nTDzmwbUNm1wVg7GZUCNAnTUDzK3mQj5wE/NcHNiJ4Bx9C1lA9m95CWh+kjMhQlw0LlFVA6dDThiF2sZggCi1XolC6MVaO2LYk0IQP23hoChJEkY9etoMuKDJchoABbFnpIkqV97CzHigwnwrgG7CyFuNUM2yQFEcSRnMj8gGoeAtOIm2WCUnJqZJjWgVQaUhUUVaG2hPUoFRADxjARnAIpKDx3dWiTSVlojbgMj/ETYNCGAMCLFKkSK3iwBKM08C1pke9rFT1xvsAR1IYB4/UWcVtz9edG0SCaJrMME5O4EAEIYohcGdZWe6fVATz9RYCbrxCaVI6BlJxRJgKcMIT9JWxwBiG3SGBW7KNuoSXYGefqJgpJsTnwIINMNYoeIhhMQA7xkCAUKIT/JW46ANBIx7rEwlNY1vvVxBpLDQDyyAPkpsOFHgMDobpwikssHt2whUI4+SyTOYgiKflJ+ZBlAyQPuew/XCheWXd4/YowlwZKtjA3B/HwGMDJkqQVeC/OaZy9SmCAh9V+WAWl0ZBYcAaeRN+/58n6IuWUuUwFZBrBZIEjBe6DtRpZsQUutPgBMEKXYx9sIRaDICWCyQOrIXmRpiFxLh9fcVIMgDxY5AWwWCDqBV6kdce9+DaAkSAKk56Upk7ZgdC0AHUttdyE/8iiil7AGkpEhCvxpmpTZqssAOgsIx2pJUy6CJmBdSJE9FnxJtq+c1RPA1IJi7GihaZHpQL10SJ/l5CsJC+WxZ/rM3llDkvUxmbTjn54AkRQdyFEioNep5nvHnUnAN0S/fB+wxSKdpoKaIIlm77GN12PLgLL4xk57Y/7S5zMbrXv/4nLqug8BkqJ8zbuQMzjVxEhfhXH0348AKQkZrcCWk2VDlL23SQ1Fd+2WLwGi4lW70oEIDfABG0RfAgqm1LZ9K0QbECGVm7vkx8hmDAHrDQ8rB/Vi05fueBzKWv5a0/SlQHGBEUeAlYQFafP1JPV0lSPst46uL2PKaenp/+iUKxgcceBf+5HoS6O10T6x86NbFew8B+PiCSjG3xIRAcV2j4N2BCQPcpek+at3h1BjpmgI/Eu5YxzzHlNuwJn12cmxDsD3JWAUWeoI/BgEFC+KNJm+54XMhgR6ulre/h6iCJP45GHeElW/RdfqmQO53rZFmHRqMexeuKuflq+0XX+dbRpSvtJ2a7+gsR/8mfEJCAag9/QnAZ0ZOAk4CeiMQGfzZwacBHRGoLP5MwNOAjoj0Nn8/w48Z50ZjVGnAAAAAElFTkSuQmCC';
