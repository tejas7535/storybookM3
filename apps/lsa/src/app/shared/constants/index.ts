export * from './language';
export * from './lubricant-type.enum';
export * from './lubrication-points.enum';
export * from './medium.enum';
export * from './optime.enum';
export * from './pages.enum';
export * from './power-supply.enum';
export * from './relubrication-interval.enum';
