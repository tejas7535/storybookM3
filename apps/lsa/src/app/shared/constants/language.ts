import { LangDefinition } from '@jsverse/transloco/lib/types';

export const LANGUAGE_STORAGE_KEY = 'language';

export const LANGUAGE_EN: LangDefinition = {
  id: 'en',
  label: 'English',
};

export const LANGUAGE_DE: LangDefinition = {
  id: 'de',
  label: 'Deutsch',
};

export const LANGUAGE_FR: LangDefinition = {
  id: 'fr',
  label: 'Français',
};

export const LANGUAGE_ES: LangDefinition = {
  id: 'es',
  label: 'Español',
};

export const LANGUAGE_IT: LangDefinition = {
  id: 'it',
  label: 'Italiano',
};

export const LANGUAGE_ZH: LangDefinition = {
  id: 'zh',
  label: '简体中文',
};

export const LANGUAGE_ZH_TW: LangDefinition = {
  id: 'zh_TW',
  label: '繁體中文',
};

export const LANGUAGE_TR: LangDefinition = {
  id: 'tr',
  label: 'Türkçe',
};

export const LANGUAGE_JA: LangDefinition = {
  id: 'ja',
  label: '日本語',
};

export const LANGUAGE_RU: LangDefinition = {
  id: 'ru',
  label: 'Русский',
};

export const LANGUAGE_PL: LangDefinition = {
  id: 'pl',
  label: 'Polski',
};

export const LANGUAGE_PT: LangDefinition = {
  id: 'pt',
  label: 'Português',
};

export const LANGUAGE_KO: LangDefinition = {
  id: 'ko',
  label: '한국인',
};

export const LANGUAGE_VI: LangDefinition = {
  id: 'vi',
  label: 'Tiếng Việt',
};

export const LANGUAGE_CS: LangDefinition = {
  id: 'cs',
  label: 'Čeština',
};

export const LANGUAGE_TH: LangDefinition = {
  id: 'th',
  label: 'ไทย',
};

export const AVAILABLE_LANGUAGES: LangDefinition[] = [
  LANGUAGE_EN,
  LANGUAGE_DE,
  LANGUAGE_ES,
  LANGUAGE_FR,
  LANGUAGE_IT,
  LANGUAGE_ZH,
  LANGUAGE_ZH_TW,
  LANGUAGE_TR,
  LANGUAGE_JA,
  LANGUAGE_RU,
  LANGUAGE_PL,
  LANGUAGE_PT,
  LANGUAGE_KO,
  LANGUAGE_VI,
  LANGUAGE_CS,
  LANGUAGE_TH,
];

export const FALLBACK_LANGUAGE: LangDefinition = LANGUAGE_EN;
