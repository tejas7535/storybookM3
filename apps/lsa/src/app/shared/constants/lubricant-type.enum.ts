export enum LubricantType {
  Arcanol = 'Arcanol',
  Grease = 'Grease',
  Oil = 'Oil',
}
