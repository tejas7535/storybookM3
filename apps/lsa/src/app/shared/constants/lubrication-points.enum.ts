export enum LubricationPoints {
  One = '1',
  TwoToFour = '2-4',
  FiveToEight = '5-8',
  EightOrMore = '>8',
}
