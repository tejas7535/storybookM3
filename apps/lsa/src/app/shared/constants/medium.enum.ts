export enum Medium {
  Grease = 'Grease',
  Oil = 'Oil',
}
