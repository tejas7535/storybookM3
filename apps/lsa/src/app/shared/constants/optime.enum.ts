export enum Optime {
  No = 0,
  Yes = 1,
  NoPreference = 2,
}
