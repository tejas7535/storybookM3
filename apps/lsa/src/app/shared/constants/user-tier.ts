export type UserTier = 'anonymous' | 'plus' | 'business';
