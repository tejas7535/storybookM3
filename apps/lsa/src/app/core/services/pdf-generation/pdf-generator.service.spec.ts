import { BehaviorSubject, of } from 'rxjs';

import { TranslocoService } from '@jsverse/transloco';
import { PowerSupply } from '@lsa/shared/constants';
import { UserTier } from '@lsa/shared/constants/user-tier.enum';
import {
  Accessory,
  Lubricator,
  RecommendationResponse,
} from '@lsa/shared/models';
import { MediasCallbackResponse } from '@lsa/shared/models/price-availibility.model';
import {
  createServiceFactory,
  mockProvider,
  SpectatorService,
} from '@ngneat/spectator/jest';

import {
  FontResolverService,
  ImageResolverService,
} from '@schaeffler/pdf-generator';

import { AddToCartService } from '../add-to-cart.service';
import { LsaFormService } from '../lsa-form.service';
import { PriceAvailabilityService } from '../price-availability.service';
import { RestService } from '../rest.service';
import { ResultInputsService } from '../result-inputs.service';
import { FormDataType, PDFGeneratorService } from './pdf-generator.service';
import { MOCK_RECOMMENDATION_FORM } from './recommendation-form.mock';

const MOCK_LUBRICATOR: Lubricator = {
  matNr: '12345',
  name: 'HIGHEND_LUBRICATOR',
  outputDiameter: 1,
  maxOperatingPressure: 100,
  noOfOutlets: 2,
  minTemp: -100,
  maxTemp: 100,
  volume: '250 ml',
  bundle: [
    {
      matnr: '12',
      fifteen_digit: '12',
      designation: 'Acc #1',
      qty: 5,
      pim_code: '12',
      class: 'Lubricators',
    },
    {
      matnr: '32',
      designation: 'Acc #2',
      fifteen_digit: '12',
      qty: 1,
      pim_code: '32',
      class: 'Cartridges',
    },
  ] as Accessory[],
  technicalAttributes: {
    func_principle: 'Piston Pump',
    dimensions: '20x20x20mm',
    voltage: '30v',
    medium_general: 'Grease',
    mounting_position: 'Arbitrary',
  },
} as unknown as Lubricator;

const MOCK_TABLE_DATA: FormDataType = {
  Lubricators: {
    '12': 5,
  },
};

const MOCK_RECOMMENDATION_RESPONSE: RecommendationResponse = {
  lubricators: {
    recommendedLubricator: MOCK_LUBRICATOR,
    minimumRequiredLubricator: MOCK_LUBRICATOR,
  },
} as RecommendationResponse;

describe('PDFGeneratorService', () => {
  let spectator: SpectatorService<PDFGeneratorService>;
  let service: PDFGeneratorService;
  const createService = createServiceFactory({
    service: PDFGeneratorService,
    providers: [
      mockProvider(TranslocoService),
      mockProvider(RestService, {
        recommendation$: new BehaviorSubject<RecommendationResponse>(
          MOCK_RECOMMENDATION_RESPONSE
        ),
      }),
      mockProvider(ImageResolverService, {
        fetchImages: jest.fn((i) => of(i)),
        fetchImageObject: jest.fn((i) => of(i)),
      }),
      mockProvider(FontResolverService, {
        fetchForLocale: jest.fn(() => of()),
      }),
      mockProvider(LsaFormService, {
        getRecommendationForm: jest.fn(() => ({
          getRawValue: jest.fn(() => MOCK_RECOMMENDATION_FORM),
        })),
      }),
      mockProvider(ResultInputsService, {
        getPipeLengthTranslation: jest.fn((a) => of(a)),
      }),
      mockProvider(AddToCartService, {
        getUserTier: jest.fn(() => UserTier.Business),
      }),
      mockProvider(PriceAvailabilityService, {
        priceAndAvailabilityResponse$: of<MediasCallbackResponse>({
          items: { '32': { available: true, price: 1, currency: 'EUR' } },
        }),
      }),
    ],
  });

  beforeEach(() => {
    spectator = createService();
    service = spectator.inject(PDFGeneratorService);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('convertPipeLengthString', () => {
    const mock = service['resultInputsService'][
      'getPipeLengthTranslation'
    ] as jest.Mock;
    mock.mockReturnValue('test');
    service['convertPipeLengthString'](0);
    expect(mock).toHaveBeenCalled();
  });

  it('getPowerSupplyTranslation', () => {
    const options = [
      PowerSupply.Battery,
      PowerSupply.NoPreference,
      PowerSupply.External,
    ];
    for (const opt of options) {
      service['getPowerSupplyTranslation'](opt);
    }
    expect(service['translocoService'].translate).toHaveBeenCalledTimes(
      options.length
    );
  });

  it('calling generate should call loadig', () => {
    const loadingSpy = jest.spyOn(service['loading$$'], 'next');
    service.generatePDF(true);
    service['recommendedLubData'].subscribe();
    expect(loadingSpy).toHaveBeenCalledWith(true);
  });

  it('setFormData should emit the data subjec5', () => {
    const dataSubj = jest.spyOn(service['tableData$$'], 'next');
    service.setFormData({} as FormDataType);
    expect(dataSubj).toHaveBeenCalled();
  });

  it('should select the proper lubricator', (done) => {
    const testResponse: RecommendationResponse = {
      lubricators: {
        minimumRequiredLubricator: { ...MOCK_LUBRICATOR, name: 'MINIMUM' },
        recommendedLubricator: { ...MOCK_LUBRICATOR, name: 'RECOMMENDATION' },
      },
    } as unknown as RecommendationResponse;

    service['recommendedLubData'].subscribe((lubricator) => {
      expect(lubricator).toMatchSnapshot();
      done();
    });
    service.setFormData(MOCK_TABLE_DATA);
    service['restService'].recommendation$.next(testResponse);
    service.generatePDF(true);
  });

  it('should format the groups for proper filtering in pdf', (done) => {
    const testResponse: RecommendationResponse = {
      lubricators: {
        minimumRequiredLubricator: { ...MOCK_LUBRICATOR, name: 'MINIMUM' },
        recommendedLubricator: { ...MOCK_LUBRICATOR, name: 'RECOMMENDATION' },
      },
    } as unknown as RecommendationResponse;

    service['productGroups$'].subscribe((groups) => {
      expect(groups).toMatchSnapshot();
      done();
    });
    service.setFormData(MOCK_TABLE_DATA);
    service['restService'].recommendation$.next(testResponse);
    service.generatePDF(true);
  });
});
