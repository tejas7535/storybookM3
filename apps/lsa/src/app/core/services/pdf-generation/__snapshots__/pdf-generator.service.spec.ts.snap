// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFGeneratorService extractDetailsTable should append ml if the volume already does not have a unit 1`] = `
Array [
  Array [
    undefined,
    "Piston Pump",
  ],
  Array [
    undefined,
    "20x20x20mm",
  ],
  Array [
    undefined,
    "100 ml",
  ],
  Array [
    undefined,
    "100 bar",
  ],
  Array [
    undefined,
    "30v",
  ],
  Array [
    undefined,
    "Grease",
  ],
  Array [
    undefined,
    undefined,
  ],
  Array [
    undefined,
    "2",
  ],
  Array [
    undefined,
    "Arbitrary",
  ],
]
`;

exports[`PDFGeneratorService extractDetailsTable should not append ml if the volume already has a unit 1`] = `
Array [
  Array [
    undefined,
    "Piston Pump",
  ],
  Array [
    undefined,
    "20x20x20mm",
  ],
  Array [
    undefined,
    "100 cmÂ³",
  ],
  Array [
    undefined,
    "100 bar",
  ],
  Array [
    undefined,
    "30v",
  ],
  Array [
    undefined,
    "Grease",
  ],
  Array [
    undefined,
    undefined,
  ],
  Array [
    undefined,
    "2",
  ],
  Array [
    undefined,
    "Arbitrary",
  ],
]
`;

exports[`PDFGeneratorService productGroups should format the groups for proper filtering in pdf 1`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "RECOMMENDATION",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;

exports[`PDFGeneratorService productGroups should format the groups for proper filtering in pdf 2`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "MINIMUM",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;

exports[`PDFGeneratorService productGroups should format the groups for proper filtering in pdf 3`] = `
Array [
  Object {
    "price": "5.00 EUR",
    "products": Array [
      Object {
        "available": true,
        "currency": "EUR",
        "description": undefined,
        "designation": "Acc #2",
        "group": "Lubricators",
        "id": "32",
        "imageUrl": "/assets/images/placeholder.png",
        "pimid": "32",
        "price": 1,
        "quantity": 5,
      },
    ],
    "title": undefined,
  },
]
`;

exports[`PDFGeneratorService productGroups should set the image to "" when there is an issue with loading the image 1`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "RECOMMENDATION",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;

exports[`PDFGeneratorService productGroups should set the image to "" when there is an issue with loading the image 2`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "MINIMUM",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;

exports[`PDFGeneratorService productGroups should set the image to "" when there is an issue with loading the image 3`] = `
Array [
  Object {
    "price": "5.00 EUR",
    "products": Array [
      Object {
        "available": true,
        "currency": "EUR",
        "description": undefined,
        "designation": "Acc #2",
        "group": "Lubricators",
        "id": "32",
        "imageUrl": "/assets/images/placeholder.png",
        "pimid": "32",
        "price": 1,
        "quantity": 5,
      },
    ],
    "title": undefined,
  },
]
`;

exports[`PDFGeneratorService should not show prices if the cart service says so 1`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "RECOMMENDATION",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;

exports[`PDFGeneratorService should not show prices if the cart service says so 2`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "MINIMUM",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;

exports[`PDFGeneratorService should not show prices if the cart service says so 3`] = `
Array [
  Object {
    "price": "5.00 EUR",
    "products": Array [
      Object {
        "available": true,
        "currency": "EUR",
        "description": undefined,
        "designation": "Acc #2",
        "group": "Lubricators",
        "id": "32",
        "imageUrl": "/assets/images/placeholder.png",
        "pimid": "32",
        "price": 1,
        "quantity": 5,
      },
    ],
    "title": undefined,
  },
]
`;

exports[`PDFGeneratorService should select the proper lubricator 1`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "RECOMMENDATION",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;

exports[`PDFGeneratorService should select the proper lubricator for recommendation false 1`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "RECOMMENDATION",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;

exports[`PDFGeneratorService should select the proper lubricator for recommendation false 2`] = `
Object {
  "bundle": Array [
    Object {
      "class": "Lubricators",
      "designation": "Acc #1",
      "fifteen_digit": "12",
      "matnr": "12",
      "pim_code": "12",
      "qty": 5,
    },
    Object {
      "class": "Cartridges",
      "designation": "Acc #2",
      "fifteen_digit": "12",
      "matnr": "32",
      "pim_code": "32",
      "qty": 1,
    },
  ],
  "matNr": "12345",
  "maxOperatingPressure": 100,
  "maxTemp": 100,
  "minTemp": -100,
  "name": "MINIMUM",
  "noOfOutlets": 2,
  "outputDiameter": 1,
  "technicalAttributes": Object {
    "dimensions": "20x20x20mm",
    "func_principle": "Piston Pump",
    "medium_general": "Grease",
    "mounting_position": "Arbitrary",
    "voltage": "30v",
  },
  "volume": "250 ml",
}
`;
